File: ninja-be/src/main/java/pl/ochnios/ninjabe/model/dtos/conversation/ConversationCriteria.java
package pl.ochnios.ninjabe.model.dtos.conversation;

import java.time.Instant;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class ConversationCriteria {

    private String globalSearch;
    private Integer minMessageCount;
    private Integer maxMessageCount;
    private Instant minCreatedAt;
    private Instant maxCreatedAt;
    private String summary;
    private String userFirstname;
    private String userLastname;
    private Boolean deleted;
}

File: samurai-fe/src/model/api/conversation/ConversationCriteria.ts
import { SearchCriteria } from "../page/SearchCriteria.ts";

export interface ConversationCriteria extends SearchCriteria {
  globalSearch?: string;
  minMessageCount?: number;
  maxMessageCount?: number;
  minCreatedAt?: string;
  maxCreatedAt?: string;
  summary?: string;
  userFullName?: string;
  deleted?: boolean;
}
